packages/@aws-cdk/aws-s3/lib/notifications-resource/notifications-resource-handler.ts

```diff
@@ -84,6 +84,8 @@ export class NotificationsResourceHandler extends cdk.Construct {
       }
     });
 
+    resource.node.addDependency(role);
```

Comment:
Similar to https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda/lib/function.ts#L458, however the InLineLambda resource already duplicates a little bit from the lambda module.